### PR TITLE
wrap incoming engine calls in batch

### PR DESCRIPTION
**Pull request by chrisnojima**
@keybase/react-hackers this wraps dispatches from the engine in react/redux's batch. I believe this will slightly reduce renders when we have a ton of incoming rpcs. I actually was hosing the app by settings breakpoints and toggling wifi on/off. SetState would overflow due to incoming rpcs and react would die. This stop that from happening also

### CODE REVIEW

**Comment by chrisnojima** [comment]
Note i think we could gain a lot by doing this w/ redux-saga but we need an api thats in 1.x

File: shared/engine/index-impl.tsx [hunk 1]
```diff
@@ -35,8 +36,31 @@ class Engine {
   _hasConnected: boolean = isMobile // mobile is always connected
   // App tells us when the sagas are done loading so we can start emitting events
   _sagasAreReady: boolean = false
+
+  static _realDispatch: Dispatch
```
Comment (buoyad) [inline shared/engine/index-impl.tsx]: Why use `static` for these?

Comment (chrisnojima) [reply shared/engine/index-impl.tsx]: its static in engine so other things can just use it and not have an instance (like session etc). Before this PR dispatch was static, now these helpers of dispatch are also

File: shared/store/configure-store.tsx [hunk 2]
```diff
@@ -76,6 +76,8 @@ const errorCatching = () => next => action => {
     if (lastError.message === error.message) {
       return
     }
+    // eslint-disable-next-line
+    debugger
```
[the remark this answers is not in the document]
Comment (chrisnojima) [reply shared/store/configure-store.tsx]: tnx

File: shared/engine/index-impl.tsx (hunk 1 above)
Comment (buoyad) [reply shared/engine/index-impl.tsx]: Ah k, afaict they're not used outside this class (session uses `deprecatedGetDispatch`). Slightly confusing.

**Pull request merged**